kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/PaymentRequestServiceImpl.java

```diff
@@ -106,6 +119,106 @@ protected PaymentMethodGeneralLedgerPendingEntryService getPaymentMethodGeneralL
+
+                    // No discount is applied to other item types other than item line
+                    // See KFSMI-5210 for details
+
+                    // total amount should be the line item total, not the grand total
+                    totalAmount = paymentRequestDocument.getLineItemTotal();
+
+                    // prorate item line accounts only
+                    Set includedItemTypeCodes = new HashSet();
+                    includedItemTypeCodes.add(PurapConstants.ItemTypeCodes.ITEM_TYPE_ITEM_CODE);
+                    includedItemTypeCodes.add(PurapConstants.ItemTypeCodes.ITEM_TYPE_SERVICE_CODE);
+
+                    summaryAccounts = purapAccountingService.generateSummaryIncludeItemTypesAndNoZeroTotals(paymentRequestDocument.getItems(), includedItemTypeCodes);
+                    //if summaryAccount is empty then do not call generateAccountDistributionForProration as
+                    //there is a check in that method to throw NPE if accounts percents == 0..
+                    //KFSMI-8487
+                    if (summaryAccounts != null) {
+                    distributedAccounts = purapAccountingService.generateAccountDistributionForProration(summaryAccounts, totalAmount, PurapConstants.PRORATION_SCALE, PaymentRequestAccount.class);
+                    }
+                    if (PurapConstants.AccountDistributionMethodCodes.SEQUENTIAL_CODE.equalsIgnoreCase(accountDistributionMethod)) {
+                        purapAccountingService.updatePreqAccountAmountsWithTotal(distributedAccounts, item.getTotalAmount());
+
+                    } else {
+
+                        boolean rulePassed = true;
+                        // check any business rules
+                        rulePassed &= kualiRuleService.applyRules(new PurchasingAccountsPayableItemPreCalculateEvent(paymentRequestDocument, item));
+
+                        if (rulePassed) {
+                            purapAccountingService.updatePreqProporationalAccountAmountsWithTotal(distributedAccounts, item.getTotalAmount());
+                        }
+                    }
+                }
+                else {
+                    PurchaseOrderItem poi = item.getPurchaseOrderItem();
+                    if ((poi != null) && (poi.getSourceAccountingLines() != null) && (!(poi.getSourceAccountingLines().isEmpty())) && (poi.getExtendedPrice() != null) && ((KualiDecimal.ZERO.compareTo(poi.getExtendedPrice())) != 0)) {
+                        // use accounts from purchase order item matching this item
+                        // account list of current item is already empty
+                        item.generateAccountListFromPoItemAccounts(poi.getSourceAccountingLines());
+                    }
+                    else {
+                    	if ((poi.getExtendedPrice() != null) && ((KualiDecimal.ZERO.compareTo(poi.getExtendedPrice())) == 0))
```

Comment:
Please put curly brace at the end of the if statement and then the in-line comments below that. Also, please check tabbing.